src/Microsoft.ML.LightGBM/LightGbmTrainerBase.cs

```diff
@@ -285,8 +285,9 @@ private CategoricalMetaData GetCategoricalMetaData(IChannel ch, RoleMappedData t
                 ch.Info("Auto-tuning parameters: " + nameof(Args.UseCat) + " = " + useCat);
             if (useCat)
             {
-                trainData.Schema.Schema.TryGetColumnIndex(DefaultColumnNames.Features, out int featureIndex);
-                MetadataUtils.TryGetCategoricalFeatureIndices(trainData.Schema.Schema, featureIndex, out categoricalFeatures);
+                var featureCol = trainData.Schema.Schema.GetColumnOrNull(DefaultColumnNames.Features);
+                ch.Assert(featureCol.HasValue);
```

Comment:
>ch.Assert(featureCol.HasValue); [](start = 16, length = 31)

Similar note, can probably just use `.Value` in the line above to clarify "yes I expect this to have a value." But, as you like.